### DISCUSSION TITLE
Integrate snapshots and rename config of vm

### DISCUSSION THREAD

**Discussion by kruscho** [abbbi/virtnbdbackup#203] [Q&A / Support] [answered]
Hi,

I am new to virtnbdbackup, played a little bit around, but have some questions.
1. I am wonder, if it is possible to integrate snapshots of the VM (i want to backup)? By doing:

`sudo virtnbdbackup -d iot -l full -o /mnt/work/vm_backup/iot`

and 

`sudo virtnbdrestore -i /mnt/work/vm_backup/iot/ -o /mnt/work/vm_machines/`

i am not able to see any snapshots. I also confirmed that by:

`sudo virsh snapshot-list iot`

2. If I restore a VM (see above), I get a "vmconfig.copy.xml" with all my configurations. Is it possible, to rename this xml during the restore? Otherwise, i will overwrite this file, when i do a bulk restore of some VMs.

Thanks

**Comment by abbbi** [comment]
Hi, Snapshots are Not Part of the backup because virtnbdbackup uses bitmaps which are Independent. Adding support for snapshots doesnt make much sense, its a different way of backup. During restore if you use option to adjust the config it will remove the rederence to configured backing Images (snapshots)

**Reply by kruscho** [reply]
Hi,
yeah, i see. You define the "snapshot/state" of a VM by using a certain point in time recovery. For me, it is easier to handle snapshots via virt-manager, but i got your point  (if I get you right :) ).

When i adjust the config, i still get a general configuration file:
`adjusted config placed in: [/mnt/work/vm_machines/vmconfig.copy.xml]`

can i adjust the filename "vmconfig.copy.xml" by any option/value?

Thanks for you response and this great tool

**Reply by abbbi** [reply]
Not at the Moment, But it would be trivial to add this feature

current workaround would be to restore to a different folder Or subfolder

**Comment by abbbi** [comment, marked as answer]
ive pushed some changes to master, now virtnbdrestore accepts option -C to set the config file name during restore.

```
 ./virtnbdrestore -i /tmp/backup -o $(mktemp -d) -cC bla.xml
[..]
Adjusted config placed in: [/tmp/tmp.BWOrJ7HLGo/bla.xml]
```
